### PR TITLE
Employee movement

### LINKED ISSUE #49: proper crowd simulation

### PR DESCRIPTION
**Pull request by mertwole**
closes #49 for now, for further improvements in queuing system open a new issue

### CODE REVIEW

File: JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs [hunk 1]
```diff
@@ -30,17 +36,10 @@ private enum State
...
         public delegate void FinishedMovingHandler();
-        public event FinishedMovingHandler OnFinishedMoving;
-
-        public void SetDestination(Vector3 target_position)
-        {
-            currentDestination = target_position;
-            _ = agent.SetDestination(currentDestination);
-            state = State.BuildingPath;
-        }
+        public event FinishedMovingHandler OnReachedNeedProvider;
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs:42]:
```suggestion
        public event Action OnReachedNeedProvider;
```

File: JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs [hunk 2]
```diff
@@ -73,18 +72,49 @@ private void Update()
...
+        }
+
         private void CorrectMovement()
         {
-            agent.speed =
+            float max_speed =
                 (1.0f - personalSpace.GetCrowdMetrics())
                 * maxVelocity
                 * maxVelocityMultiplierByEffects;
+            max_speed = Mathf.Clamp(max_speed, 0.0f, float.MaxValue);
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs:115]:
```suggestion
            max_speed = Mathf.Max(max_speed, 0.0f);
```

File: JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs [hunk 3]
```diff
@@ -73,18 +72,49 @@ private void Update()
...
+
+            agent.speed = max_speed;
 
             Vector3 steering = personalSpace.GetPreferredSteeringNormalized();
             if (steering.sqrMagnitude > 0.0001)
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs:120]: What is this constant `0.0001`?

File: JamGame/Assets/Scripts/Location/EmployeeManager.cs [hunk 4]
```diff
@@ -81,20 +81,24 @@ public Result AddEmployee(EmployeeConfig config)
 
             MeetingRoomPlaces meeting_room_places =
                 DataProviderServiceLocator.FetchDataFromSingleton<MeetingRoomPlaces>();
-            NeedProvider place = meeting_room_places
-                .Places.Where(place => place.TryTake(employee))
-                .FirstOrDefault();
+            _ = new List<NeedProvider>();
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Location/EmployeeManager.cs:84]:
```suggestion
```

File: JamGame/Assets/Scripts/Location/NeedProvider.cs [hunk 5]
```diff
@@ -141,5 +250,20 @@ public void UnregisterModifier(NeedModifiers modifiers)
                 Debug.LogWarning("Modifiers to unregister not found");
             }
         }
+
+        private void OnDrawGizmos()
+        {
+            if (waitingLine.Count != 0)
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Location/NeedProvider.cs:256]:
```suggestion
            if (waitingLine.Any())
```

**Review by Kiuh — APPROVED** [review]

File: JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs (hunk 3 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Employee/Controller/ControllerImpl.cs:120]: Just some small enough number as I'm not shure that float.EPSILON will work correctly in this context

**Pull request merged**